src/platform/ESP32/Logging.cpp

```diff
@@ -26,20 +26,35 @@ void ENFORCE_FORMAT(3, 0) LogV(const char * module, uint8_t category, const char
     snprintf(tag, sizeof(tag), "chip[%s]", module);
     tag[sizeof(tag) - 1] = 0;
 
-    char formattedMsg[CHIP_CONFIG_LOG_MESSAGE_MAX_SIZE];
-    vsnprintf(formattedMsg, sizeof(formattedMsg), msg, v);
-
     switch (category)
     {
     case kLogCategory_Error:
-        ESP_LOGE(tag, "%s", formattedMsg);
+        if (esp_log_default_level >= ESP_LOG_ERROR)
+        {
+            printf(LOG_COLOR_E "E");                        // set color
```

Comment:
Can you explain why you are using printf here? This change is breaking us so I want to better understand why you are making these changes here